src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php

```diff
@@ -99,9 +109,51 @@ public function registerContainerConfiguration(LoaderInterface $loader)
                 $kernelDefinition->addTag('kernel.event_subscriber');
             }
 
-            $this->configureContainer($container, $loader);
-
             $container->addObjectResource($this);
+            $container->fileExists($this->getProjectDir().'/config/bundles.php');
+            $container->setParameter('container.dumper.inline_class_loader', \PHP_VERSION_ID < 70400 || !ini_get('opcache.preload'));
+            $container->setParameter('container.dumper.inline_factories', true);
+            $container->setParameter('kernel.secret', '%env(APP_SECRET)%');
+
+            try {
+                $this->configureContainer($container, $loader);
+
+                return;
+            } catch (\TypeError $e) {
+                $file = $e->getFile();
+
+                if (0 !== strpos($e->getMessage(), sprintf('Argument 1 passed to %s::configureContainer() must be an instance of %s,', static::class, ContainerConfigurator::class))) {
+                    throw $e;
+                }
+            }
+
+            $defaultDefinition = (new Definition())->setAutowired(true)->setAutoconfigured(true);
+            $kernelLoader = $loader->getResolver()->resolve($file);
+            $kernelLoader->setCurrentDir(\dirname($file));
+            $instanceof = &\Closure::bind(function &() { return $this->instanceof; }, $kernelLoader, $kernelLoader)();
```

Comment:
Is it possible for there to be any `instanceof` at this point?

Reply:
No, but the point here is to get a reference, so it doesn't matter.
Because the property is protected, it is part of the BC promise so we can this this without breaking any contracts.